File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        # i=0
        # og_head=head
        # while head:
        #     i+=1
        #     head=head.next
        # head=og_head
        # i=i-n
        # j=0
        # if i<0: return og_head
        # if i==0: return og_head.next
        # while head.next:
        #     j+=1
        #     if j==i and j!=0:
        #         head.next=head.next.next
        #         return og_head
        #     head=head.next        
        dummy=ListNode(0,head)
        left=dummy
        right=head
        while n>0:
            right=right.next
            n-=1
        while right:
            left=left.next
            right=right.next
        left.next=left.next.next
        return dummy.next
        
        

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode dummy = new ListNode(0, head);
        ListNode left = dummy, right = findnode(head, n);
        
        while (right != null){
            left = left.next;
            right = right.next;
        }
        left.next = left.next.next;
        return dummy.next;
    }
    private ListNode findnode(ListNode cur, int k){
        while ((cur != null) && (k > 0)){
            cur = cur.next;
            k--;
        }
        return cur;
    }
}